google-cloud-kubernetes/deploy.py

```diff
@@ -0,0 +1,163 @@
+                        action='store_true')
+    parser.add_argument(
+        '-c', '--config', type=str, help='Configuration file')
+    parser.add_argument('-n', '--new', help='Create a new cluster',
+                        action='store_true')
+
+    args = parser.parse_args()
+    print(args)
+
+    if args.quite:
+        verbose = False
+
+    if not args.config:  # No configuration file given, check config.json
+        info('Loading cloud configuration from "config.json"')
+        with open('config.json', 'rb') as fd:
+            cloud_config = json.load(fd)
+    else:
+        info('Loading cloud configuration from %s' % args.config)
+        with open(args.config, 'rb') as fd:
+            cloud_config = json.load(fd)
+
+    parser_config = {
+        'cloud-config': cloud_config,
+        'create-cluster': args.new
+    }
+
+    return parser_config
+
+
+def clone_project(user):
+    git_url = 'https://github.com/%s/loklak_server.git' % user
+    info('Cloning %s...' % git_url)
+    output = subprocess.check_output(['git', 'clone', git_url, '-q'])
+    info('Successfully cloned %s- \n%s' % (git_url, output))
+    os.chdir('loklak_server')
+
+
+def checkout_branch(branch):
+    output = subprocess.check_output(['git', 'checkout', branch])
+    info(output)
+
+
+def build_docker_image(project, version):
+    tag = 'gcr.io/%s/loklak:%s' % (project, version)
+    command = 'docker build -t %s .' % tag
+    info('Building image with TAG %s' % tag)
+    output = subprocess.check_output(command.split())
+    info('Docker build complete - \n%s' % output)
+
+
+def push_docker_image(project, version):
+    tag = 'gcr.io/%s/loklak:%s' % (project, version)
+    info('Pushing docker image to GCR...')
+    output = subprocess.check_output(['gcloud', 'docker', '-- push', tag])
+    info('Pushed %s to GCR - \n%s' % (tag, output))
+
+
+def create_cluster(cluster_name, num_nodes, machine_type, zone):
+    info('Creating Cluster...')
+    output = subprocess.check_output(
+        ['gcloud', 'container', 'clusters', 'create', cluster_name,
+         '--num-nodes', num_nodes, '--machine-type', machine_type,
+         '--zone', zone])
+    info('Cluster created successfully - \n%s' % output)
+
+
+def run_deployment(deployment, project, version):
+    tag = 'gcr.io/%s/loklsk:%s' % (project, version)
+    info('Running deployment %s using tag %s ' % (deployment, tag))
+    output = subprocess.check_output(['kubectl', 'run', deployment,
+                                      '--image=%s' % tag, '--port=80'])
+    info(output)
+
+
+def expose_deployment(deployment):
+    info('Exposing deployment %s' % deployment)
+    output = subprocess.check_output(['kubectl', 'expose', 'deployment',
+                                      deployment, '--type=LoadBalancer'])
+    info(output)
+
+
+def fetch_public_ip(deployment):
+    info('Polling to get public IP for loklak')
+    regex_ip = re.compile(r'%s[\t\ ]+[0-9]+\.[0-9]+\.[0-9]+\.[0-9]'
```

Comment:
@vibhcool: I'm sorry but I didn't get the API that you are talking about. Could you please share a link or something?